src/amo/components/Collection/index.js

```diff
@@ -102,6 +104,31 @@ export type SaveAddonNoteFunc = (
   notes: string,
 ) => void;
 
+// This gets the new collection page.
+const getNewCollectionPage = (collection) => {
+  const { numberOfAddons, pageSize } = collection;
+
+  if (pageSize) {
+    return Math.ceil((numberOfAddons - 1) / pageSize) || 1;
+  }
+
+  return 1;
+};
+
+const shouldChangePage = (collection) => {
+  const { numberOfAddons, pageSize } = collection;
+
+  let page = 1;
+
+  if (pageSize) {
+    page = (numberOfAddons - 1) / pageSize;
```

Comment:
I see that we do that in the function above. Can we do merge the two functions?

```js
export const computeNewCollectionPage = (
  collection: CollectionType,
): number | null => {
  const { numberOfAddons, pageSize } = collection;

  let page = 1;
  if (pageSize) {
    page = (numberOfAddons - 1) / pageSize;
  }

  if (!Number.isInteger(page)) {
    return null;
  }

  return Math.ceil(page) || 1;
};
```

Reply:
In which case do we have `page` that is not an integer?

Reply:
So, I think we can do this:

```js
export const computeNewCollectionPage = (
  collection: CollectionType,
): number => {
  const { numberOfAddons, pageSize } = collection;

  let page = 1;
  if (pageSize) {
    page = Math.ceil((numberOfAddons - 1) / pageSize);
  }

  return page || 1;
};


```

Reply:
and use it like this:

```js
  removeAddon: RemoveCollectionAddonFunc = (addonId: number) => {
    const { collection, dispatch, errorHandler, filters, history } = this.props;

    invariant(collection, 'collection is required');

    const { slug, authorUsername: username } = collection;

    invariant(slug, 'slug is required');
    invariant(username, 'username is required');

    let { page } = filters;
    let shouldPushNewRoute = false;
    const newCollectionPage = computeNewCollectionPage(collection);

    if (page !== newCollectionPage) {
      page = newCollectionPage;
      shouldPushNewRoute = true;
    }

    dispatch(
      removeAddonFromCollection({
        addonId,
        errorHandlerId: errorHandler.id,
        filters: {
          ...filters,
          page,
        },
        slug,
        username,
      }),
    );

    if (shouldPushNewRoute) {
      const { location } = history;

      history.push({
        pathname: location.pathname,
        query: {
          ...location.query,
          page,
        },
      });
    }
  };

```

What do you think? That would prevent a push to happen for no reason, and also make the code more understandable, no?

Reply:
unless I a missing something, the only difference I really see is this piece:

```javascript
  let { page } = filters;

    if (isNewPage) {
      page = getNewCollectionPage(collection);
    }
...
```

As far as pushing the page, I was already doing that - only pushing if isNewPage was true but I agree your names make it a little clearer

Reply:
Exactly! You did the hard work by finding the issue and the patch to fix it. I just made it a bit more readable because I really had troubles understanding it.

Reply:
and my comments too : )